packages/react/src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -206,7 +208,7 @@ const HierarchyList = ({
 }) => {
   const [expandedIds, setExpandedIds] = useState(defaultExpandedIds);
   const [searchValue, setSearchValue] = useState('');
-  const [filteredItems, setFilteredItems] = useState(cloneDeep(items));
+  const [filteredItems, setFilteredItems] = useState(() => cloneDeep(items));
```

Comment:
This is very smart!

Reply:
I definitely think it's good to instantiate the state this way but what is the reason for the cloneDeep? Especially now.

Reply:
This is to keep the prop items intact as they'll get mutated a bunch inside of the component

Reply:
Removed as its actually not needed